src/component/forgotpassword/__tests__/ForgotPassword.test.tsx

```diff
@@ -0,0 +1,325 @@
+        />
+      </MemoryRouter>
+    );
+    const usernameInput = () => wrapper.find('input[name="username"]');
+    changeInputValue(usernameInput(), INVALID_EMAIL);
+    usernameInput().simulate("keyPress", { key: "Enter" });
+    expect(usernameInput().props().value).toEqual(INVALID_EMAIL);
+    expect(requestPasswordReset).not.toHaveBeenCalled();
+  });
+
+  it("renders alert with message", async () => {
+    const requestResult = {
+      error: {
+        messageId: undefined,
+        message: "Error message text",
+      },
+      status: AsyncActionStatus.FAILURE,
+      type: MessageType.ERROR,
+    } as AsyncFailureAction;
+    requestPasswordReset = jest
+      .fn()
+      .mockImplementation(() => Promise.resolve(requestResult));
+    const wrapper = mountWithIntl(
+      <MemoryRouter>
+        <ForgotPassword
+          loading={false}
+          requestPasswordReset={requestPasswordReset}
+          {...intlFunctions()}
+        />
+      </MemoryRouter>
+    );
+
+    const usernameInput = () => wrapper.find('input[name="username"]');
+    const alert = () => wrapper.find(Alert);
+    changeInputValue(usernameInput(), VALID_EMAIL);
+
+    // required to wait on .then execution on requestPasswordReset
+    // and to let component handle its state
+    await act(async () => {
+      usernameInput().simulate("keyPress", { key: "Enter" });
+      await new Promise(process.nextTick);
+      wrapper.update();
+    });
+
+    expect(alert().props().color).toEqual(requestResult.type);
+    expect(alert().text()).toEqual(requestResult.error.message);
+  });
+
+  it("renders alert with internationalized messageId", async () => {
+    const requestResult: AsyncFailureAction = {
+      error: new ErrorInfo(MessageType.ERROR, {
+        messageId: "forgotPassword.username.notValidEmail",
+        message: "Error message text",
+      }),
+      status: AsyncActionStatus.FAILURE,
+      type: MessageType.ERROR,
+    };
+    requestPasswordReset = jest
+      .fn()
+      .mockImplementation(() => Promise.resolve(requestResult));
+    const wrapper = mountWithIntl(
+      <MemoryRouter>
+        <ForgotPassword
+          loading={false}
+          requestPasswordReset={requestPasswordReset}
+          {...intlFunctions()}
+        />
+      </MemoryRouter>
+    );
+
+    const usernameInput = () => wrapper.find('input[name="username"]');
+    const alert = () => wrapper.find(Alert);
+    changeInputValue(usernameInput(), VALID_EMAIL);
+
+    // required to wait on .then execution on requestPasswordReset
+    // and to let component handle its state
+    await act(async () => {
+      usernameInput().simulate("keyPress", { key: "Enter" });
+      await new Promise(process.nextTick);
+      wrapper.update();
+    });
+
+    expect(alert().props().color).toEqual(requestResult.type);
+    expect(alert().text()).toEqual(
+      i18n("forgotPassword.username.notValidEmail")
+    );
+  });
+
+  it("renders alert on success", async () => {
```

Comment:
Don't use alerts for this. Use the `Messages` component and corresponding message actions.